File: services/dotnet-microservices/ProductService/Domain/Models/Product.cs
namespace ProductService.Domain.Models;

public record Product(
    Guid Id,
    string Name,
    string? Description,
    decimal Price,
    Guid? CategoryId,
    string? ImageUrl,
    bool IsActive,
    DateTime CreatedAt,
    DateTime UpdatedAt
);

public record Category(
    Guid Id,
    string Name,
    string? Description,
    Guid? ParentId,
    DateTime CreatedAt
);

public record Review(
    Guid Id,
    Guid ProductId,
    Guid UserId,
    int Rating,
    string? Comment,
    DateTime CreatedAt
);

public record CreateProductRequest(
    string Name,
    string? Description,
    decimal Price,
    Guid? CategoryId,
    string? ImageUrl
);

public record UpdateProductRequest(
    string? Name,
    string? Description,
    decimal? Price,
    Guid? CategoryId,
    string? ImageUrl,
    bool? IsActive
);

public record CreateCategoryRequest(string Name, string? Description, Guid? ParentId);

public record CreateReviewRequest(Guid UserId, int Rating, string? Comment);

File: services/nodejs-microservices/product-service/src/domain/models/Product.ts
/**
 * Product Domain Models
 */
export interface Product {
  id: string;
  name: string;
  description?: string;
  price: number;
  categoryId?: string;
  imageUrl?: string;
  status: ProductStatus;
  createdAt: Date;
  updatedAt: Date;
}

export enum ProductStatus {
  ACTIVE = 'ACTIVE',
  INACTIVE = 'INACTIVE',
  DISCONTINUED = 'DISCONTINUED',
}

export interface Category {
  id: string;
  name: string;
  description?: string;
  parentId?: string;
  createdAt: Date;
}

export interface ProductReview {
  id: string;
  productId: string;
  userId: string;
  rating: number;
  comment?: string;
  createdAt: Date;
}
